engine/sql/src/main/java/io/deephaven/engine/sql/Sql.java

```diff
@@ -83,18 +82,16 @@ private static Map<String, Table> currentScriptSessionNamedTables() {
         final Map<String, Table> scope = new HashMap<>();
         // getVariables() is inefficient
         // See SQLTODO(catalog-reader-implementation)
-        for (Entry<String, Object> e : currentScriptSession().getVariables().entrySet()) {
-            if (e.getValue() instanceof Table) {
-                scope.put(e.getKey(), (Table) e.getValue());
+        QueryScope queryScope = ExecutionContext.getContext().getQueryScope();
+        for (String name : queryScope.getParamNames()) {
+            Object paramValue = queryScope.unwrapObject(queryScope.readParamValue(name));
```

Comment:
To make the unwrapping safe? At least not for the implementation (only python) we have today, no.

Reply:
No, to make the `readParamValue` results be consistent with `getParamNames`. Otherwise, you could have a `MissingVariableException` due to concurrent mutation of the `QueryScope`.
Feels like a missing method to get values, to be honest.